lifecycle/lifecycle-livedata-core-ktx-lint/src/test/java/androidx/lifecycle/lint/NonNullableMutableLiveDataDetectorTest.kt

```diff
@@ -760,4 +765,40 @@ Fix for src/com/example/MyClass2.kt line 30: Change `LiveData` type to nullable:
             ).indented()
         ).expectClean()
     }
+
+    // https://issuetracker.google.com/issues/184830262
+    @Test
+    fun justKotlinObject() {
+        check(
+            kotlin(
+                """
+                package com.example
+
+                object Foo
+            """
+            ).indented()
+        ).expectClean()
+    }
+
+    @Test
+    fun genericParameterDefinition() {
+        check(
+            kotlin(
+                """
+                package com.example
+
+                import androidx.lifecycle.MutableLiveData
+
+                class Foo<T>(
+                    var target: MutableLiveData<T>
+                ) {
+                
+                    fun foo(value: T) {
+                        target.value = value
```

Comment:
Should this not be testing `target.value = null`?

Reply:
value T is already nullable by default because <T> is defined without any type information. https://kotlinlang.org/docs/generics.html#upper-bounds

Reply:
Right, but we already know a value with type T will match just fine, what we actually want to verify here is that the lint rule will ignore setting a variable with a type of T to null since T is indeed nullable.

Reply:
Test is updated.